### ISSUE TITLE
Code deletes plugins when `--extensions-dir` argument used

### ISSUE THREAD

**Issue by colin-grant-work** [microsoft/vscode#175037]
Does this issue occur when all extensions are disabled?: Yes/No

No - relates to behavior of plugins.

- VS Code Version: 1.75.1
- OS Version: RHEL 7

Steps to Reproduce:

> Because I'm not very familiar with the profile system, there are a few variables of whose significance I'm unsure. The basic scenario is that my team maintains a folder with development versions of unpublished plugins for testing purposes. After upgrading from a version of VSCode with no profiles (1.71.0) to one with profiles (1.75.1), starting the application with `--extensions-dir=<our demo directory>` results in the deletion(!!) of all of the plugins that were in that folder and the creation of an empty `extensions.json` (just `[]`).

1. Place (unpublished - not sure if it matters) plugins in a folder.
2. With no profiles set up...
3. Start VS Code with `--extensions-dir=<my folder>`
4. Find that VS Code loads with no plugin shown as installed.
5. Check the directory and find that everything has been deleted, and in their place is an empty `extensions.json` file.

**Comment by sandy081** [comment]
Sorry about what happened. Is it possible to reproduce it consistently? If so, can you please share the following log when it happens?

- F1 > Open View... > Shared

**Comment by colin-grant-work** [comment]
It is consistently reproducible, but I'm afraid the logs for the Shared output channel aren't very illuminating. After starting a session, I've checked to confirm that the contents of the folder specified by the `--extensions-dir` argument have been deleted, but the only thing in the Shared output channel is this:

```log
2023-02-22 08:26:19.584 [info] Starting worker process with pid 6936 (type: fileWatcher, window: 1).
```

**Comment by sandy081** [comment]
Are you running the command when there is an existing VS Code instance?

**Comment by colin-grant-work** [comment]
No, definitely cleaning up in between sessions. I have deleted my `~/.config/Code` and `~/.vscode` folders, and now I'm not able to reproduce the deletion behavior (though I've achieved rather the opposite of profiles' intended functionality: rather than preserving my settings across machines, they've made me delete them all). What I can still reproduce consistently is that having a profile makes the application ignore the `--extensions-dir` argument.

1. Start VSCode with extensions dir in some workspace.
2. Create a profile from that session's settings (Profiles: Create from Current Profile)
3. Close VSCode and start a new session, passing the `--extensions-dir` flag.
4. VSCode will ignore (though now not delete - progress) the plugins in the specified directory.

It's a design decision, but I think that anything specified explicitly for the current session should override anything intended to be shared: the opposite of the priority expressed here:

https://github.com/microsoft/vscode/blob/b438c3854c2541f349b99e94411f1d91186827ae/src/vs/platform/extensionManagement/common/extensionsScannerService.ts#L196
